src/js/jsx/sections/style/Type.jsx

```diff
@@ -63,36 +63,55 @@ define(function (require, exports, module) {
                 return collection.pluck(document.layers.selected, "opacity");
             };
 
+            if (!Immutable.is(this.state.postScriptMap, nextState.postScriptMap)) {
+                if (nextState.postMap) {
+                    // The list of all selectable type faces
+                    var typefaces = nextState.postScriptMap
+                        .entrySeq()
+                        .sortBy(function (entry) {
+                            return entry[0];
+                        })
+                        .map(function (entry) {
+                            var psName = entry[0],
+                                fontObj = entry[1];
+
+                            return {
+                                id: psName,
+                                title: fontObj.font
+                            };
+                        })
+                        .toList();
+
+                    this.setState({
+                        typefaces: typefaces
```

Comment:
I don't have a good advice on where to put this, `componentWillReceiveProps` only relies on `nextProps`. Feels like this logic should be part of the fontStore, instead of this view doing it.